### ISSUE TITLE
Opening a file with mimeapp creates a child process.

### ISSUE THREAD

**Issue by kaldyr** [sxyazi/yazi#194]
When opening a file (with "o") with a mimeapp (zathura or any other application), the resulting process is a child process of yazi.  Closing yazi will also close the child process.  It would be nice to pop into yazi, open a file, and close yazi without also closing the spawned process.

**Comment by sxyazi** [comment]
Done by https://github.com/sxyazi/yazi/pull/216, please give it a try!

**Comment by kaldyr** [comment]
Just compiled from the pr-5202b9db branch and set the orphan = true and it works perfectly!  Thanks!

**Comment by kaldyr** [comment]
Just compiled from the pr-5202b9db branch and set the orphan = true and it works perfectly! Thanks!

**Comment by sxyazi** [comment]
Thank you for testing, merged!

**Comment by Brixy** [comment]
Hi guys, I don’t want to clutter this closed thread; but `orphan = true` has no effect at all for me. I compiled yazi from the current main branch and from the corresponding commit.

Setup: Void Linux, glibc, wayland, river.

Also roughly tested yazi under xorg + i3 to ensure that wayland does not cause the problem.

Is this a yazi bug on linux, and is there anything I can do to help you test this? Am I just missing anyething?

Thanks again for yazi and your excellent work!

**Comment by sxyazi** [comment]
Hi @Brixy, could you please paste your opener configuration here, and the steps you followed? This will help me understand what happened.

**Comment by kaldyr** [comment]
It may be worth adding more information to the --version flag to figure out specifically which build is currently being executed.

**Comment by kaldyr** [comment]
Just confirmed working on [492929a](https://github.com/sxyazi/yazi/commit/492929a0b82f593c886fce3c533575b96489d1ab).

in my flake.nix inputs:
```
yazi.url = "github:sxyazi/yazi";
```

in my home-manager.nix:
```
    programs.yazi = {
      enable = true;
      package = inputs.yazi.packages.${pkgs.system}.yazi;
      enableFishIntegration = true;
      settings = {
        manager = {
          sort_by = "natural";
          sort_reverse = false;
          sort_dir_first = true;
          show_hidden = false;
          show_symlink = true;
        };
        opener = {
          audio = [{
            exec = "clapper \"$@\"";
            orphan = true;
          }];
          video = [{
            exec = "clapper \"$@\"";
            orphan = true;
          }];
          folder = [{
            exec = "$EDITOR \"$@\"";
            block = true;
          }];
          text = [{
            exec = "$EDITOR \"$@\"";
            block = true;
          }];
          fallback = [
            {
              exec = "xdg-open \"$@\"";
              orphan = true;
            }
            {
              exec = "$EDITOR \"$@\"";
              block = true;
            }
          ];
        };
      };
    };
```

which generates the following ~/.config/yazi/yazi.toml file:
```
[manager]
show_hidden = false
show_symlink = true
sort_by = "natural"
sort_dir_first = true
sort_reverse = false

[opener]
[[opener.audio]]
exec = "clapper \"$@\""
orphan = true

[[opener.fallback]]
exec = "xdg-open \"$@\""
orphan = true

[[opener.fallback]]
block = true
exec = "$EDITOR \"$@\""

[[opener.folder]]
block = true
exec = "$EDITOR \"$@\""

[[opener.text]]
block = true
exec = "$EDITOR \"$@\""

[[opener.video]]
exec = "clapper \"$@\""
orphan = true
```

**Comment by Brixy** [comment]
Thanks a lot for your replies!

Here’s my steps to reproduce:

I just deleted my `target/` folder and compiled from [the current commit](https://github.com/sxyazi/yazi/commit/492929a0b82f593c886fce3c533575b96489d1ab). Functions _newer_ than the the `orphan` functionality work, so the newest version of yazi should be compiled correctly.

Moreover, I switched back to the default settings and just edited a single line for the `video` opener in `yazi.toml`:

```
{ exec = 'mpv "$@"', orphan = true },
```

When I open a video it seems to be run as a child process anyway, and it is listed in the list of tasks (`w`).

As soon as yazi is closed (`q`, `^q` or through the window manager) the mpv window is closed, too.

Thanks a lot for investigating!

**BTW:** Just noticed that `undo` does not seem to work, either. Please let me know if I should open an issue for this.

**Comment by sxyazi** [comment]
I followed the steps above to reproduce the issue and found that the reason is that `mpv` is not only a GUI program but also a CLI.

It retrieves `stdout` while displaying the GUI, and when Yazi exits, `stdout` becomes inactive, causing `mpv` to exit as well.

Although `mpv --no-terminal` can make it work, in case there are similar programs, I have created a PR to fix it. Please give it a try. https://github.com/sxyazi/yazi/pull/229

**Comment by sxyazi** [comment]
> It may be worth adding more information to the --version flag to figure out specifically which build is currently being executed.

Seems like this is what https://github.com/sxyazi/yazi/pull/206 to do

**Comment by Brixy** [comment]
Excellent. Can confirm that [#229](https://github.com/sxyazi/yazi/pull/229) works perfectly—with `mpv` and “real” GUIs (tested with `abiword`).

So mpv was either the best or worst example ;-)

Thank you very much for all your work!

Just as a side note: When yazi is killed by just closing its window using the window manager (river) the mpv window is closed, too. That’s no problem at all. Yet, when an `nnn` window is closed this way the video window remains open.

**Comment by sxyazi** [comment]
> When yazi is killed by just closing its window using the window manager (river) the mpv window is closed, too.

I've tried for a long time but couldn't fix it. I've decided to just leave it here. If anyone knows the reason, please let me know :)

> BTW: Just noticed that undo does not seem to work, either. Please let me know if I should open an issue for this.

I think what you're asked is the undo function in the Vi-like input component. https://yazi-rs.github.io/docs/usage/configuration/keymap#normal-mode

**Comment by Brixy** [comment]
> I think what you're asked is the undo function in the Vi-like input component.

Yes, sorry. I did not investigate here. I thought it was possible to e.g. undo file deletion or mass renaming.

> I've tried for a long time but couldn't fix it. I've decided to just leave it here. If anyone knows the reason, please let me know :)

Thank you very much for investigating!!

**Comment by sxyazi** [comment]
> Just as a side note: When yazi is killed by just closing its window using the window manager (river) the mpv window is closed, too. That’s no problem at all. Yet, when an nnn window is closed this way the video window remains open.

Hi @Brixy, I have revisited it and found a solution. A PR is created should fix it: https://github.com/sxyazi/yazi/pull/290

**Comment by Brixy** [comment]
> I have revisited it and found a solution. A PR is created should fix it: #290

Thank you very much for your brilliant work! I can confirm that this works.

---

BTW: I found a use case where this is important:

If you opened yazi ‘directly’ e.g. using the following command in a window manager, closing yazi using `q` would also have killed the terminal emulator and yazi’s orphans.

```
riverctl map normal Super Y spawn 'foot -D $HOME -e yazi'
```

With your changes everything works perfectly. Thanks a lot for this!
